app/views/teachers/TeacherStudentView.coffee

```diff
@@ -0,0 +1,392 @@
+    return if @destroyed # Don't do anything if page was destroyed after db request
+    @updateLastPlayedString()
+    @updateLevelProgressMap()
+    @updateLevelDataMap()
+
+
+  # onCourseInstancesSync: ->
+  #   # @sessions = new CocoCollection([], { model: LevelSession })
+  #   for courseInstance in @courseInstances.models
+  #     sessions = new CocoCollection([], { url: "/db/course_instance/#{courseInstance.id}/level_sessions", model: LevelSession })
+  #     @supermodel.loadCollection(sessions, { data: { project: ['level', 'playtime', 'creator', 'changed', 'state.complete'].join(' ') } })
+  #     courseInstance.sessions = sessions
+  #     sessions.courseInstance = courseInstance
+  #     courseInstance.sessionsByUser = {}
+  #     @listenToOnce sessions, 'sync', (sessions) ->
+  #       @sessions.add(sessions.slice())
+  #       for courseInstance in @courseInstances.models
+  #         courseInstance.sessionsByUser = courseInstance.sessions.groupBy('creator')
+  #
+  #   # Generate course instance JIT, in the meantime have models w/out equivalents in the db
+  #   for course in @courses.models
+  #     query = {courseID: course.id, classroomID: @classroom.id}
+  #     courseInstance = @courseInstances.findWhere(query)
+  #     if not courseInstance
+  #       courseInstance = new CourseInstance(query)
+  #       @courseInstances.add(courseInstance)
+  #       courseInstance.sessions = new CocoCollection([], {model: LevelSession})
+  #       # sessions.courseInstance = courseInstance
+  #       courseInstance.sessionsByUser = {}
+
+  updateLastPlayedString: ->
+    # Make sure all our data is loaded, @sessions may not even be intialized yet
+    return unless @courses.loaded and @levels.loaded and @sessions?.loaded and @user?.loaded
+
+    # Use lodash to find the last session for our user, @sessions already sorted by changed date
+    session = _.findLast @sessions.models, (s) => s.get('creator') is @user.id
+
+    return unless session
+
+    # Find course for this level session, for it's name
+    # Level.original is the original id, used for level versioning, and connects levels to level sessions
+    for versionedCourse in @classroom.get('courses') ? []
+      for level in versionedCourse.levels
+        if level.original is session.get('level').original
+          # Found the level for our level session in the classroom versioned courses
+          # Find the full course so we can get it's name
+          course = _.find @courses.models, (c) => c.id is versionedCourse._id
+          break
+
+    # Find level for this level session, for it's name
+    level = @levels.findWhere({original: session.get('level').original})
+
+    # Update last played string based on what we found
+    @lastPlayedString = ""
+    @lastPlayedString += course.get('name') if course
+    @lastPlayedString += ": " if course and level
+    @lastPlayedString += level.get('name') if level
+    @lastPlayedString += ", on " if @lastPlayedString
```

Comment:
Hm, I might not be clear on what you mean, but if seems as though we'd only want a colon between the course and level name if both exist; otherwise, just display the course or the level without a joining ":".